### PR TITLE
ENCD-5783-deprecrated-files-crashing-file-details

### PR DESCRIPTION
**Pull request by zoldello**

### CODE REVIEW

File: src/encoded/static/components/filegallery.js [hunk 1]
```diff
@@ -3034,14 +3044,19 @@ class FileGalleryRendererComponent extends React.Component {
...
-                ? this.getBrowserSelectedIndex()
-                : dropdown ? dropdown.selectedIndex : null;
+            let selectedIndex = 0;
+            let compiledAnalysis = compileAnalyses(context, allFiles, 'choose analysis');
+
+            if (currentTab === 'browser') {
+                compiledAnalysis = compiledAnalysis.filter((c) => {
+                    const analysisFiles = filterForVisualizableFiles(c.files.map((f) => allFiles.find((aF) => aF['@id'] === f)));
+                    return analysisFiles.length > 0;
+                });
```
Comment (forresttanaka) [inline src/encoded/static/components/filegallery.js]: Seems like an exact copy of `this.getBrowserAnalysis()`. You couldn’t call that here?

**Review by forresttanaka — CHANGES_REQUESTED** [review]
Looks like the functionality change is `compileAnalysis` gets called regardless of the tab, but the rest is the same?

File: src/encoded/static/components/filegallery.js (hunk 1 above)
Comment (zoldello) [reply src/encoded/static/components/filegallery.js]: Corrected.

**Review by forresttanaka — APPROVED** [review]
Looks much simpler; thanks!

**Pull request merged**